pkg/chunked/filesystem_linux.go

```diff
@@ -458,10 +458,15 @@ func safeMkdir(dirfd int, mode os.FileMode, name string, metadata *fileMetadata,
 		defer parentFile.Close()
 		parentFd = int(parentFile.Fd())
 	}
-
-	if err := unix.Mkdirat(parentFd, base, uint32(mode)); err != nil {
-		if !os.IsExist(err) {
-			return &fs.PathError{Op: "mkdirat", Path: name, Err: err}
+	// It can happen if name is a path in a form like "./".  When it happens, set base to point to
+	// the current directory.
+	if base == "" {
+		base = "."
+	} else {
+		if err := unix.Mkdirat(parentFd, base, uint32(mode)); err != nil {
```

Comment:
⚠️  AFAICS with `r.Name == "../..`, this ends up doing `Mkdirat(dirfd, "..")`.

That’s, luckily, guaranteed to fail with `EEXIST`, so I don’t think there is any actual vulnerability, but it shows that the logic is imprecise. This, similarly, applies to some other code here, I think, and probably needs a universal solution.

And below, we are trying to `setFileAttrs` on `dirfd/..` — in that case I open `openFileUnderRoot` prevents that from breaking out.

Reply:
yes `openFileUnderRoot` prevents opening files above dirfd.

I've modified the code to prevent using "..", even if it is not a security problem